### PR TITLE
[CI][TensorFlow] make bazel rules compatible with different settings

### PR DESCRIPTION
**Pull request by Orion34-lanbo**
1. add rule for `libiomp5.so`
2. depend `@local_config_cuda//cuda:cublasLt` only when `if_has_cublaslt` is true make it works for cu100 or cpu device
3. link myelin if myelin libs exist

### CODE REVIEW

File: tensorflow_blade/src/tensorrt/BUILD [hunk 1]
```diff
@@ -57,6 +58,9 @@ cc_library(
         "//src/util:logging",
     ] + if_has_cublaslt([
         "@local_config_cuda//cuda:cublasLt",
+    ]) + if_has_myelin([
+        "@local_config_cuda_static//:nvrtc",
```
Comment (qiuxiafei) [inline tensorflow_blade/src/tensorrt/BUILD:62]: why is nvrtc bind with myelin ?

File: third_party/bazel/cuda_static/cuda_static.BUILD.tpl [hunk 2]
```diff
@@ -21,3 +21,8 @@ cc_import(
     name = "cudnn_static",
     static_library = "lib64/libcudnn_static.a",
 )
+
+cc_import(
+    name = "nvrtc",
+    shared_library = "lib64/libnvrtc.so",
```
Comment (qiuxiafei) [inline third_party/bazel/cuda_static/cuda_static.BUILD.tpl:27]: an shared lib in a build file named static...

File: third_party/bazel/tensorrt/repo.bzl [hunk 3]
```diff
@@ -1,14 +1,66 @@
...
+    has_myelin = files_exist(
+            repo_ctx,
+            [
+                "lib/libmyelin_compiler_static.a",
+                "lib/libmyelin_executor_static.a",
+                "lib/libmyelin_pattern_library_static.a",
+                "lib/libmyelin_pattern_runtime_static.a",
+            ]
+    )
+    if_has_myelin = has_myelin[0] and has_myelin[1] and has_myelin[2] and has_myelin[3]
```
Comment (qiuxiafei) [inline third_party/bazel/tensorrt/repo.bzl:57]: pls use `all`: https://docs.bazel.build/versions/main/skylark/lib/globals.html#all

File: third_party/bazel/tf/tf_configure.bzl [hunk 4]
```diff
@@ -44,15 +44,15 @@ def _tf_configure_impl(repository_ctx):
         "-DTF_MAJOR={}".format(tf_major),
         "-DTF_MINOR={}".format(tf_minor),
     ]
-    if tf_is_pai:
+    if tf_is_pai == "True":
```
Comment (qiuxiafei) [inline third_party/bazel/tf/tf_configure.bzl:47]: looks weird ... why not making `tf_is_pai` a boolean variable, and use `str(tf_is_pai)` for templating.

File: third_party/bazel/cuda_static/cuda_static.BUILD.tpl (hunk 2 above)
Comment (Orion34-lanbo) [reply third_party/bazel/cuda_static/cuda_static.BUILD.tpl:27]: I did not find a better for this supplementary cuda rule, may cuda_supplement?

File: tensorflow_blade/src/tensorrt/BUILD (hunk 1 above)
Comment (Orion34-lanbo) [reply tensorflow_blade/src/tensorrt/BUILD:62]: For trt7, if nvrtc is not linked when myelin libs are linked, we will encounter undefined symbol issue, this symbol is from nvrtc. However for later trt versions, such as trt8, myelin libs no longer exists, we don't need to link nvrtc

File: third_party/bazel/tensorrt/repo.bzl (hunk 3 above)
Comment (Orion34-lanbo) [reply third_party/bazel/tensorrt/repo.bzl:57]: Originally, I used `all`, another bug makes me think `all` doesn't work as expected.

File: third_party/bazel/tf/tf_configure.bzl (hunk 4 above)
Comment (Orion34-lanbo) [reply third_party/bazel/tf/tf_configure.bzl:47]: `tf_is_pai` is acquired from env as string use `get_env_bool_value_str` func, maybe using `if bool(tf_is_pai)` here is ok.

File: third_party/bazel/cuda_static/cuda_static.BUILD.tpl (hunk 2 above)
Comment (Orion34-lanbo) [reply third_party/bazel/cuda_static/cuda_static.BUILD.tpl:27]: done

File: third_party/bazel/tensorrt/repo.bzl (hunk 3 above)
Comment (Orion34-lanbo) [reply third_party/bazel/tensorrt/repo.bzl]: done

File: third_party/bazel/tf/tf_configure.bzl (hunk 4 above)
Comment (Orion34-lanbo) [reply third_party/bazel/tf/tf_configure.bzl]: done

**Review by qiuxiafei — APPROVED** [review]

**Pull request merged**